### PR TITLE
Feature TR-4685 NRPS membership difference

### PR DESCRIPTION
**Pull request by wazelin**
# [TR-4685](https://oat-sa.atlassian.net/browse/TR-4685)

https://user-images.githubusercontent.com/2943256/203092331-3c24acf6-2f5e-4fce-825c-426226b86360.mov

## How to test

1. Create a custom NRPS membership
2. Call the platform's NRPS membership endpoint
3. Verify the Link header has the `rel="difference"` value
4. Call that link, make sure it returns an empty membership set
5. Update the original membership
6. Call the link again
7. Make sure the differences are displayed

This PR aims to bring [Membership Differences](https://www.imsglobal.org/spec/lti-nrps/v2p0#membership-differences) to the NRPS implementation in DevKit.

This feature is required to re-submit for the LTI Advantage Platform certification.

The validation with the IMS' test tool [went well](https://oat-sa.atlassian.net/wiki/spaces/NEX/pages/1899266885/1EdTech+former+IMS+certification+procedure+LTI+1.3+platform#Submitting-the-results), will merge once the final confirmation from IMS has been received.

### CODE REVIEW

File: src/Action/Platform/Nrps/EditMembershipAction.php [hunk 1]
```diff
@@ -84,7 +80,16 @@ public function __invoke(Request $request, string $membershipIdentifier): Respon
         }
 
         if ($membership->getMembers()->count() !== 0) {
-            $members = json_encode($membership->getMembers());
+            $members = json_encode(
+                array_values(
+                    array_filter(
+                        $membership->getMembers()->all(),
+                        static function (MemberInterface $member) {
+                            return $member->getStatus() !== MemberInterface::STATUS_DELETED;
```
Comment (kilatib) [inline src/Action/Platform/Nrps/EditMembershipAction.php:88]: why you takes only deleted here?

File: src/Action/Platform/Nrps/EditMembershipAction.php [hunk 2]
```diff
@@ -54,28 +54,24 @@ class EditMembershipAction
...
         $this->twig = $twig;
         $this->factory = $factory;
         $this->router = $router;
-        $this->memberFactory = $memberFactory;
     }
 
     public function __invoke(Request $request, string $membershipIdentifier): Response
     {
-        $membership = $this->repository->find($membershipIdentifier);
+        $membership = $this->service->findMembership($membershipIdentifier);
```
Comment (kilatib) [inline src/Action/Platform/Nrps/EditMembershipAction.php:74]: Here you takes only 
`MemberInterface::STATUS_ACTIVE, MemberInterface::STATUS_INACTIVE`

statuses end then filter deletes, is it ok?

Comment (kilatib) [inline src/Action/Platform/Nrps/EditMembershipAction.php:74]:
```suggestion
        $membership = $this->service->findMembership($membershipIdentifier, [MemberInterface::STATUS_DELETED]);
```

**Review by poyuki — APPROVED** [review]
- [ ] New code is covered by tests (if applicable)
- [ ] Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x] New code is respecting code style rules
- [x] New code is respecting best practices
- [x] New code is not subject to concurrency issues (if applicable)
- [x] Feature is working correctly on my local machine (if applicable)
- [x] Acceptance criteria are respected
- [x] Pull request title and description are meaningful
- [x] Pull request's target is not `master`
- [x] Commits are following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Changelog is updated according to changes (if applicable)
- [ ] Documentation is updated according to changes (if applicable)

File: src/Action/Platform/Nrps/EditMembershipAction.php (hunk 1 above)
Comment (wazelin) [reply src/Action/Platform/Nrps/EditMembershipAction.php:88]: That filters _not_ deleted ones.

File: src/Action/Platform/Nrps/EditMembershipAction.php (hunk 2 above)
Comment (wazelin) [reply src/Action/Platform/Nrps/EditMembershipAction.php:74]: No, that retrieves _all_ records regardless of their state if I'm not mistaken.

Comment (wazelin) [reply src/Action/Platform/Nrps/EditMembershipAction.php:74]: That would fetch only deleted ones. I don't think this would be intentional.

**Pull request merged**